### PR TITLE
Fix chain cash regression (do not modify on transfer)

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

File: pallets/cash/src/pipeline.rs [hunk 1]
```diff
@@ -369,28 +369,30 @@ fn prepare_augment_cash<T: Config>(
...
     let total_cash_post = st
         .get_total_cash_principal::<T>()
         .sub(recipient_repay_principal)
         .map_err(|_| Reason::InsufficientChainCash)?;
 
     st.set_cash_principal::<T>(recipient, recipient_cash_post);
     st.set_total_cash_principal::<T>(total_cash_post);
-    st.set_chain_cash_principal::<T>(chain_id, chain_cash_principal_post);
+
+    if from_external {
```
Comment (hayesgm) [inline pallets/cash/src/pipeline.rs:388]: Are we allowed to bail after we set things here? Is that behavior used elsewhere?

**Review by hayesgm — APPROVED** [review]
Looks good; I think it's relatively clean and fixes a key bug. Quick question in re: semi-stateful effects.

**Review by hayesgm — APPROVED** [review]
lgtm

File: pallets/cash/src/pipeline.rs (hunk 1 above)
Comment (jflatow) [reply pallets/cash/src/pipeline.rs:388]: We don't write until commit right?

Comment (jflatow) [reply pallets/cash/src/pipeline.rs:388]: Or not sure what you mean

**Pull request merged**